pkg/statsd/handler_cloud.go

```diff
@@ -145,25 +152,32 @@ func (ch *CloudHandler) RunMetrics(ctx context.Context, statser stats.Statser) {
 		case <-ctx.Done():
 			return
 		case <-ticker.C:
-			ch.emit(statser)
+			select {
```

Comment:
Why not `select` on `<-ctx.Done()` and `ch.emitChan <- statser`? I.e. why blocking here is bad?

Reply:
This is (sort of) the model of the metrics-refactor branch, where a channel tells you when to emit and it drops it if you miss your shot.  For now this is just a quick'n'dirty to get things in.